src/auto/enums.rs

```diff
@@ -494,3 +494,113 @@ impl FromGlib<glib_sys::GTimeType> for TimeType {
         }
     }
 }
+
+#[cfg(any(feature = "v2_66", feature = "dox"))]
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Clone, Copy)]
+#[non_exhaustive]
+pub enum UriError {
+    Failed,
+    BadScheme,
+    BadUser,
+    BadPassword,
+    BadAuthParams,
+    BadHost,
+    BadPort,
+    BadPath,
+    BadQuery,
+    BadFragment,
+    #[doc(hidden)]
+    __Unknown(i32),
+}
+
+#[cfg(any(feature = "v2_66", feature = "dox"))]
+impl fmt::Display for UriError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "UriError::{}",
+            match *self {
+                UriError::Failed => "Failed",
+                UriError::BadScheme => "BadScheme",
+                UriError::BadUser => "BadUser",
+                UriError::BadPassword => "BadPassword",
+                UriError::BadAuthParams => "BadAuthParams",
+                UriError::BadHost => "BadHost",
+                UriError::BadPort => "BadPort",
+                UriError::BadPath => "BadPath",
+                UriError::BadQuery => "BadQuery",
+                UriError::BadFragment => "BadFragment",
+                _ => "Unknown",
+            }
+        )
+    }
+}
+
+#[cfg(any(feature = "v2_66", feature = "dox"))]
+#[doc(hidden)]
+impl ToGlib for UriError {
+    type GlibType = glib_sys::GUriError;
+
+    fn to_glib(&self) -> glib_sys::GUriError {
+        match *self {
+            UriError::Failed => glib_sys::G_URI_ERROR_FAILED,
+            UriError::BadScheme => glib_sys::G_URI_ERROR_BAD_SCHEME,
+            UriError::BadUser => glib_sys::G_URI_ERROR_BAD_USER,
+            UriError::BadPassword => glib_sys::G_URI_ERROR_BAD_PASSWORD,
+            UriError::BadAuthParams => glib_sys::G_URI_ERROR_BAD_AUTH_PARAMS,
+            UriError::BadHost => glib_sys::G_URI_ERROR_BAD_HOST,
+            UriError::BadPort => glib_sys::G_URI_ERROR_BAD_PORT,
+            UriError::BadPath => glib_sys::G_URI_ERROR_BAD_PATH,
+            UriError::BadQuery => glib_sys::G_URI_ERROR_BAD_QUERY,
+            UriError::BadFragment => glib_sys::G_URI_ERROR_BAD_FRAGMENT,
+            UriError::__Unknown(value) => value,
+        }
+    }
+}
+
+#[cfg(any(feature = "v2_66", feature = "dox"))]
+#[doc(hidden)]
+impl FromGlib<glib_sys::GUriError> for UriError {
+    fn from_glib(value: glib_sys::GUriError) -> Self {
+        match value {
+            0 => UriError::Failed,
+            1 => UriError::BadScheme,
+            2 => UriError::BadUser,
+            3 => UriError::BadPassword,
+            4 => UriError::BadAuthParams,
+            5 => UriError::BadHost,
+            6 => UriError::BadPort,
+            7 => UriError::BadPath,
+            8 => UriError::BadQuery,
+            9 => UriError::BadFragment,
+            value => UriError::__Unknown(value),
+        }
+    }
+}
+
+#[cfg(any(feature = "v2_66", feature = "dox"))]
+impl ErrorDomain for UriError {
+    fn domain() -> Quark {
+        unsafe { from_glib(glib_sys::g_uri_quark()) }
```

Comment:
This looks wrong. That should be `g_uri_error_quark()`. The above function does not exist.

It's also wrong in the GLib annotations (CC @elmarco )

Reply:
Can we handle this from our side until a fix lands upstream?

Reply:
I think only be changing the `.gir` file unfortunately.

Reply:
I.e. something in `fix.sh` in `gir-files`. No need to regenerate everything else for that, just glib is enough.

Reply:
It's the error quark string. I don't think we can change it now.

Reply:
```xml
    <enumeration name="UriError"
                 version="2.66"
                 c:type="GUriError"
                 glib:error-domain="g-uri-quark">
```

This should match with the function for getting the quark AFAIU. There should be an annotation to override that in the enum definition.

Reply:
I think this is actually a bug in `gir`. This is the quark string, not a function. So far there was also always a function with that same name though.

I would suggest adjusting the .gir file for now and then creating an issue for fixing `gir` to handle this correctly. Should maybe just create a `once_cell::Sync::Lazy<glib::Quark>` from that static string instead.